File: internal/testdata/release-smoke/rails-safe/controller.rb
# Rails safe fixture: parameterized ActiveRecord queries.
# The safe pattern "where(" should suppress PF-RAILS-SQLI-003 -IP false positives
# on parameterized ORM queries.
class UsersController < ApplicationController
  def index
    # SAFE: parameterized ActiveRecord where with hash syntax
    @users = User.where(name: params[:name])
    render json: @users
  end

  def show
    # SAFE: parameterized ActiveRecord find_by
    @user = User.find_by(id: params[:id])
    render json: @user
  end
end

File: internal/testdata/release-smoke/symfony-safe/Controller.php
<?php
// Symfony safe fixture: parameterized Doctrine query with setParameter.
// The safe pattern "setParameter|->where\\(" should suppress
// PF-SYMFONY-SQLI-002 and TP-PHP001 -IP false positives.

namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;
use Doctrine\ORM\EntityManagerInterface;

class UserController extends AbstractController
{
    public function index(Request $request, EntityManagerInterface $em)
    {
        $name = $request->query->get('name');
        // SAFE: Doctrine DQL with setParameter
        $users = $em->createQuery('SELECT u FROM App\Entity\User u WHERE u.name = :name')
            ->setParameter('name', $name)
            ->getResult();
        return new Response(json_encode($users));
    }
}
